web/app/js/components/CheckModal.jsx

```diff
@@ -0,0 +1,311 @@
+            return <PriorityHigh className={`${classes.icon} ${classes.iconWarning}`} fontSize="small" />;
+          default:
+            null;
+        }
+      })()}
+    </React.Fragment>
+  );
+};
+
+Icon.propTypes = {
+  classes: PropTypes.shape({}).isRequired,
+  type: PropTypes.oneOf(['success', 'error', 'warning']).isRequired,
+};
+
+class CheckModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleOpenChange = this.handleOpenChange.bind(this);
+    this.runCheck = this.runCheck.bind(this);
+    this.handleApiError = this.handleApiError.bind(this);
+
+    this.state = {
+      open: false,
+      running: false,
+      success: undefined,
+      results: {},
+      error: undefined,
+    };
+  }
+
+  handleOpenChange = () => {
+    this.setState(prevState => {
+      return {
+        open: !prevState.open,
+      };
+    });
+  }
+
+  runCheck = () => {
+    this.setState({ running: true });
+
+    this.props.api.setCurrentRequests([this.props.api.fetchCheck()]);
+    this.serverPromise = Promise.all(this.props.api.getCurrentPromises())
+      .then(([response]) => {
+        this.setState({
+          running: false,
+          open: true,
+          success: response.success,
+          results: response.results,
+          error: undefined,
+        });
+      })
+      .catch(this.handleApiError);
+  }
+
+  handleApiError = error => {
+    this.setState({
+      running: false,
+      error: error,
+    });
+  }
+
+  render() {
+    const { open, running, success, results, error } = this.state;
+    const { classes, fullScreen } = this.props;
+
+    return (
+      <React.Fragment>
+        <Grid
+          container
+          direction="row"
+          justify="center"
+          alignItems="center"
+          spacing={24}>
+          <Grid item>
+            <div className={classes.wrapper}>
+              <Button
+                variant="outlined"
+                color="primary"
+                disabled={running}
+                onClick={this.runCheck}>
+                Run Linkerd Check
+              </Button>
+
+              { running && <CircularProgress size={24} className={classes.buttonSpinner} /> }
+            </div>
+          </Grid>
+        </Grid>
+
+        { this.state.error !== undefined && <ErrorBanner message={error} /> }
+
+        <Dialog
+          className={classes.dialog}
+          open={open}
+          scroll="paper"
+          TransitionComponent={Transition}
+          fullScreen={fullScreen}
+          maxWidth="md"
+          fullWidth
+          keepMounted
+          onClose={this.handleOpenChange}>
+          <DialogTitle>
+            <Grid
+              container
+              direction="row"
+              justify="space-between"
+              alignItems="center">
+              <Grid item>
+                Linkerd Check
+              </Grid>
+
+              {success !== undefined &&
+                <Grid item>
+                  <Chip
```

Comment:
I was wondering if I can add a new not required prop to `util/Chip.jsx` for the icon to reuse this component on this PR.

Reply:
Hi @cynthia-sg, what is the unrequired prop you would add to the `Chip` component? Can you paste in an example of the diffed code?

Reply:
One approach could be this: 

```
import Chip from '@material-ui/core/Chip';
import PropTypes from 'prop-types';
import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  good: {
    color: theme.status.dark.good,
    border: '1px solid ' + theme.status.dark.good
  },
  warning: {
    color: theme.status.dark.warning,
    border: '1px solid ' + theme.status.dark.warning,
  },
  bad: {
    color: theme.status.dark.danger,
    border: '1px solid ' + theme.status.dark.danger,
  }
});

function SimpleChip(props) {
  const { classes, label, type, icon } = props;

  return (
    <Chip
      className={classes[type]}
      label={label}
      icon={icon}
      variant="outlined" />
  );
}

SimpleChip.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  icon: PropTypes.element,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

SimpleChip.defaultProps = {
  icon: undefined,
};

export default withStyles(styles, { withTheme: true })(SimpleChip);
```

Another approach to use always same icons could be: 

```
import Check from '@material-ui/icons/Check';
import Chip from '@material-ui/core/Chip';
import Close from '@material-ui/icons/Close';
import PriorityHigh from '@material-ui/icons/PriorityHigh';
import PropTypes from 'prop-types';
import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  good: {
    color: theme.status.dark.good,
    border: '1px solid ' + theme.status.dark.good
  },
  warning: {
    color: theme.status.dark.warning,
    border: '1px solid ' + theme.status.dark.warning,
  },
  bad: {
    color: theme.status.dark.danger,
    border: '1px solid ' + theme.status.dark.danger,
  },
  icon: {
    border: 0,
    marginLeft: "10px",
  },
});

const Icon = ({type, classes}) => {
  const className = `${classes[type]} ${classes.icon}`;

  return (
    <React.Fragment>
      {(() => {
        switch (type) {
          case "good":
            return <Check className={className} fontSize="small" />;
          case "bad":
            return <Close className={className} fontSize="small" />;
          case "warning":
            return <PriorityHigh className={className} fontSize="small" />;
          default:
            null;
        }
      })()}
    </React.Fragment>
  );
};

Icon.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  type: PropTypes.string.isRequired,
};

function SimpleChip(props) {
  const { classes, label, type, icon } = props;

  return (
    <Chip
      className={classes[type]}
      label={label}
      icon={icon ? <Icon classes={classes} type={type} /> : undefined}
      variant="outlined" />
  );
}

SimpleChip.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  icon: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

SimpleChip.defaultProps = {
  icon: false,
};

export default withStyles(styles, { withTheme: true })(SimpleChip);
```